PR 리뷰 반영

randychoi95 opened the pull request:
- placeName을 넘겨줄 때, 값을 한 번만 읽고 deinit되기 때문에 stream(CurrentPublisher)을 제거하고 String을 넘겨주게 수정
- UI Component를 private으로 수정
- ViewController, UITableViewCell에서 Entities 의존성을 제거하고 ViewModel을 추가하여 ViewModel을 이용하여 UI 설정
- Proxypackage에서 Formatter 객체를 생성하고, 여러 곳에서 재사용하게 만듦

randychoi95 committed:
:refactor: placeName을 사용하기 위한 리팩토링

- annotation을 선택할 때, 장소 이름을 넘겨줌
- PlaceListRepository에 장소 이름 저장

randychoi95 committed:
:lipstick: PlaceListViewController

- headerView 구성
- PlaceListUsecase를 통해서 장소 목록 가져온 후 UI update하는 로직 추가

randychoi95 committed:
:sparkles: MyLocation to PlaceList DataStream

- Combine의 Publisher를 custom하여 여러 Riblet에 전달할 수 있게 구현
- 빌더를 통해 "장소이름" 전달
- PlaceListUsecase 및 PlaceRepository 조회 메소드 수정

randychoi95 committed:
:lipstick: configure uitableview

- uitableview / uitableview cell 추가

f-lab-max commented on PlaceStory/Domain/Sources/Repositories/PlaceListRepository.swift:
```diff
@@ -10,7 +10,7 @@ import Entities
 import Foundation
 
 public protocol PlaceListRepository {
-    func fetchPlaceRecordFrom(userId: String, placeName: String) -> AnyPublisher<[PlaceRecord], Error>
```
👍👍

f-lab-max commented on PlaceStory/MyLocation/Sources/MyLocation/MyLocationInteractor.swift:
```diff
@@ -34,6 +34,7 @@ protocol MyLocationInteractorDependency {
     var locationServiceUseCase: LocationServiceUseCase { get }
     var mapServiceUseCase: MapServiceUseCase { get }
     var appSettingsServiceUseCase: AppSettingsServiceUseCase { get }
+    var placeListUsecase: PlaceListUsecase { get }
```
쓰지 않는 의존성은 삭제

f-lab-max commented on PlaceStory/MyLocation/Sources/MyLocation/MyLocationInteractor.swift:
```diff
@@ -141,8 +142,9 @@ final class MyLocationInteractor: PresentableInteractor<MyLocationPresentable>,
-    func didSelectAnnotationView() {
-        router?.attachPlaceList()
+    func didSelectAnnotationView(_ placeName: String) {
+        let placeNamePublisher = CurrentPublisher(placeName)
```
stream을 쓸때는 자식이 계속 살아있으면서 바뀌는 값에 subscribing 해야할 때이고, PlaceList처럼 값을 한번만 읽고 deinit 되고 새로운 PlaceList가 생성될때는 그냥 string만 1회성으로 넘겨주면 됩니다.

f-lab-max commented on PlaceStory/PlaceDiary/Sources/PlaceList/PlaceListInteractor.swift:
```diff
@@ -5,36 +5,67 @@
         super.didBecomeActive()
-        // TODO: Implement business logic here.
+        
+        presenter.configureTitle(from: dependency.placeNamePublisher.currentValue)
```
자식이 CurrentPublisher를 받으면, 자식도 send를 할 수 있기 때문에 안됩니다. 읽을수만 있는 read-only 객체로 넘겨주세요.

f-lab-max commented on PlaceStory/PlaceDiary/Sources/PlaceList/Cell/PlaceRecordTableViewCell.swift:
```diff
@@ -0,0 +1,168 @@
+final class PlaceRecordTableViewCell: UITableViewCell {
+    
+    // MARK: - UI Components
+    let cellContentView: UIView = {
```
커스텀의 서브뷰들은 private으로 만들어주고, update(viewModel: ViewModel) 같은 퍼블릭 메서드 하나로 뷰를 설정할 수 있게 해주세요

f-lab-max commented on PlaceStory/PlaceDiary/Sources/PlaceList/Cell/PlaceRecordTableViewCell.swift:
```diff
@@ -0,0 +1,168 @@
+    }
+
+    func configureUI(
+        _ placeRecord: PlaceRecord
```
커스텀 뷰 레벨에서 도메인객체를 바로 쓰지 않고, interactor에서 domain -> 뷰모델로 변환후에 뷰에서는 굉장히 심플하게 뷰모델 값을 서브뷰에 연결해주는 역할만 하게 바꿔주세요

f-lab-max replied:
뷰모델은 뷰의 소유 "너가 나를 쓰고 싶어? 그러면 내가 선언한 뷰모델에 맞춰서 데이터를 가공해서 내놔"

f-lab-max commented on PlaceStory/PlaceDiary/Sources/PlaceList/Cell/PlaceRecordTableViewCell.swift:
```diff
@@ -0,0 +1,168 @@
+import SnapKit
+import Utils
+import UIKit
+import Entities
```
이거 제거가 목표

f-lab-max commented on PlaceStory/PlaceDiary/Sources/PlaceList/Cell/PlaceRecordTableViewCell.swift:
```diff
@@ -0,0 +1,168 @@
+        
+        placeRecordContentLabel.text = placeRecord.recordDescription
+        
+        let dateFormatter = DateFormatter()
```
DateFormatter는 생성이 비싼 객체입니다. 가급적 포맷당 하나만 만들어두고 재사용 바랍니다.

f-lab-max reviewed (changes requested):
제환님이 앞으로 코딩하면서 집중적으로 신경써야할거:
객체에 너무 과도한 책임/기능을 부여하지 않는지.
필요없는 것에 의존하고 있지는 않은지.

f-lab-max commented on PlaceStory/PlaceDiary/Sources/PlaceList/PlaceListViewController.swift:
```diff
@@ -5,7 +5,9 @@
 //  Created by 최제환 on 1/30/24.
 //
 
+import Entities
```
가급적 뷰컨/뷰에는 도메인에 의존하지 않게.
뷰모델을 선언해서 쓰기

randychoi95 committed:
:refactor: PR 리뷰 반영

Commit:
:sparkles: Attach PlaceRecordEditor Riblet

- PlaceListViewController의 addButton을 탭하면 PlaceRecordEditor Riblet Attach
- 취소/완료 버튼 탭, modal close 할 때 PlaceRecordEditor Riblet Detach

randychoi95 committed:
Merge branch 'main' into feature/configure-PlaceListViewController
